app/presenters/tree_builder.rb

```diff
@@ -351,16 +351,16 @@ def x_build_node(object, pid, options)    # Called with object, tree node parent
        options[:open_all] ||
        object[:load_children] ||
        node[:expand]
-      kids = x_get_tree_objects(object, options, nil, parents).each_with_object([]) do |o, acc|
-        acc.concat(x_build_node(o, node[:key], options))
+      kids = x_get_tree_objects(object, options, parents).map do |o|
+        x_build_node(o, node[:key], options)
       end
       node[:children] = kids unless kids.empty?
     else
-      if x_get_tree_objects(object, options.merge(:count_only => true), nil, parents) > 0
+      if x_get_tree_objects(object, options.merge(:count_only => true), parents) > 0
         node[:isLazy] = true  # set child flag if children exist
       end
     end
-    [node]
+    node
```

Comment:
Yes, exactly. This change is the reason. Probably it's more clear if you look at separate commits rather than the whole diff at once.